nav2_costmap_2d/src/costmap_2d_ros.cpp

```diff
@@ -435,6 +435,13 @@ void Costmap2DROS::resetLayers()
 bool Costmap2DROS::getRobotPose(geometry_msgs::msg::PoseStamped & global_pose) const
 {
   tf2::toMsg(tf2::Transform::getIdentity(), global_pose.pose);
+
+  if (robot_base_frame_ == global_frame_) {
```

Comment:
Isn't this check a little strange? When would someone be checking for the robot pose of `base_link` relative to `base_link`? Poses in the costmap sense are relative to the global frames which should be globally fixed like `map`, or `odom`

Reply:
This type of check is done in various places in the DWB code. I think the idea is that the caller is configurable. It can be configured to have any global  frame and any local frame. If they are configured the same ... well, it doesn't matter. The code still works.